snuba/clickhouse/query_dsl/accessors.py

```diff
@@ -0,0 +1,72 @@
+from functools import reduce
+from typing import List, Optional, Sequence, Set
+
+from snuba.clickhouse.query import Query
+from snuba.query.types import Condition
+from snuba.util import is_condition
+
+
+def get_project_ids_in_query(query: Query, project_column: str) -> Optional[Set[int]]:
+    """
+    Finds the project ids this query is filtering according to the legacy query
+    representation.
+
+    It returns a set of referenced project_ids if relevant conditions are found or
+    None if no project_id condition is found in the query. Empty set means that multiple
+    conflicting project id conditions were found.
```

Comment:
In the case of empty set, are we assuming something must be wrong (and not that the user actually sent conflicting project id conditions) and so we apply final just in case?

Reply:
We are assuming the user sent conflicting conditions. While the query, from the product standpoint would be nonsensical, it is still a valid query. Nothing is really wrong for what concerns this method. 
If we suspected that was a false negative of our condition we should probably apply final to be safe, but there are no false negative in that case, if this method identifies conditions that are conflicting, they are such (possible bugs apart).